src/include/migraphx/op/group_query_attention.hpp

```diff
@@ -0,0 +1,588 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_GROUP_QUERY_ATTENTION_HPP
+#define MIGRAPHX_GUARD_OPERATORS_GROUP_QUERY_ATTENTION_HPP
+
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/shape_for_each.hpp>
+#include <migraphx/par_for.hpp>
+#include <migraphx/gemm.hpp>
+#include <migraphx/argument.hpp>
+#include <fstream>
+#include <iostream>
+#include <iomanip>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct gqa_parameters
+{
+    std::size_t batch_size;           // Batch size used by input
+    std::size_t sequence_length;      // Sequence length used by input
+    std::size_t hidden_size;          // Hidden size used by input
+    std::size_t head_size;            // Head size
+    std::size_t rotary_embedding_dim; // Rotary embedding dimension.
+    std::size_t num_heads;            // num_heads = hidden_size / head_size
+    std::size_t max_sequence_length;  // Sequence length used by cos/sin cache
+    std::size_t head_stride;          // Head stride
+    std::size_t seq_stride;           // Sequence stride
+    std::size_t batch_stride;         // Batch stride
+    std::size_t position_ids_format;  // Format of position ids - 0 is (1), 1 is (batch_size,
```

Comment:
Why not make this a `bool`? And perhaps name it as `position_ids_use_batch` to make it clearer.